cmd/registry/main.go

```diff
@@ -67,6 +68,15 @@ func main() {
 	handler = panicHandler(handler)
 	handler = gorhandlers.CombinedLoggingHandler(os.Stdout, handler)
 
+	if config.HTTP.Secret == "" {
+		var secretBytes [32]byte
+		if _, err := rand.Read(secretBytes[:]); err != nil {
```

Comment:
This is a little subtle: you need to use `ioutil.ReadAll` here or it has to retry on error.